contributors/design-proposals/customresources-validation.md

```diff
@@ -0,0 +1,464 @@
+
+The table below shows the cases when server-side and client-side validation methods are applicable.
+
+| Case                                               | Server-Side  | Client-Side   |
+|:--------------------------------------------------:|:------------:|:-------------:|
+| Kubectl create/edit/replace with validity feedback | &#x2713;     | &#x2717;      | 
+| Custom controller creates/updates CRs              | &#x2713;     | &#x2717;      |
+| CRs are created by an untrusted party              | &#x2713;     | &#x2717;      |
+| Not making validation for CRs a special case       | &#x2713;     | &#x2717;      |
+
+The above table is an evidence that we need server-side validation as well, next to the client-side validation we easily get, nearly for free, by serving Swagger/OpenAPI specs in apiextension-apiserver. This is especially true in situations when CRs are used by components that are out of the control of the admin. Example: A user can create a database CR for a Database-As-A-Service. In this case, only server-side validation can give confidence that the CRs are well formed.
+
+### <a name="schema-change"></a>Existing Instances and changing the Schema
+
+If the schema is made stricter later, the existing CustomResources might no longer comply with the spec. This will make them unchangeable and essentially read-only. To avoid this, it is the responsibility of the user to make sure that any changes made to the schema are such that the existing CustomResources remain validated. 
+ 
+Note: This is the same behavior that we require for native resources. Validation cannot be made stricter in later Kubernetes versions without breaking compatibility.
+
+### <a name="sub-resources"></a>Outlook to Status Sub-Resources
+
+As another most-wanted feature, a Status sub-resource might be proposed and implemented for CRDs. The JSON-Schema proposed here might as well cover the Status field of a CR. For now this is not handled or validated in a particular way.
+ 
+When the Status sub-resource exists some day, the /status endpoint will receive a full CR object. We can then validate the whole object against the status.
+Hence, this proposal is compatible with a later sub-resource extension.
+
+### <a name="webhooks"></a>Outlook Admission Webhook
+
+Apiextensions-apiserver uses the normal REST endpoint implementation and only customizes the registry and the codecs. The admission plugins are inherited from the kube-apiserver (when running inside of it via apiserver delegation) and therefore they are supposed to apply to CRs as well. The webhook admission prototyped at https://github.com/kubernetes/kubernetes/pull/46316 is expected to work with CRs out of the box. Hence, for more advanced validation the webhook admission is an option as well (when it is merged). Clearly, providing such a webhook is more complex and therefore cannot replace a built-in JSON-Schema validation and its use-cases.
```

Comment:
better add line-breaks in those paragraph at a reasonable column number. Make it easier to comment.